pkg/daemon/handler.go

```diff
@@ -14,9 +14,9 @@ type secretServiceServer struct {
 	store SecretStore
 }
 
-func NewSecretServiceHandler() (string, http.Handler) {
+func NewSecretServiceHandler(store SecretStore) (string, http.Handler) {
```

Comment:
I chose to extract the store creation, so that the `SecretServiceHandler` is more "generic" and does not need to know which underlying secret store implementation is used. Also, this way further functions can be called on the secret store before using it in the `servicehandler` (e.g., `PopulateStore`)